central/clusterinit/backend/backend.go

```diff
@@ -52,14 +52,26 @@ func newBackend(store store.Store, certProvider certificate.Provider) Backend {
 // CheckAccess returns nil if requested access level is granted in context.
 func CheckAccess(ctx context.Context, access storage.Access) error {
 	// we need access to the API token and service identity resources
-	scopes := [][]sac.ScopeKey{
-		{sac.AccessModeScopeKey(access), sac.ResourceScopeKey(resources.APIToken.GetResource())},
+	serviceIdentityScopes := [][]sac.ScopeKey{
```

Comment:
Instead of these dances, what about the following:
```
func CheckAccess(ctx context.Context, access storage.Access) error {
	var authz authz.Authorizer
	
	switch access {
	case storage.Access_READ_ACCESS:
		authz = and.And(
			user.With(permissions.View(resources. ServiceIdentity)),
			user.With(permissions.View(resources. APIToken)))
	case storage.Access_READ_WRITE_ACCESS:
		authz = and.And(
			user.With(permissions.Modify(resources. ServiceIdentity)),
			user.With(permissions.Modify(resources. APIToken)))
	}

	// fullMethodName must be ignored by both `and.And` and `user.With`.
	return authz.Authorized(ctx, "")
}
```
I think it's safe to assume that `fullMethodName` will be ignore by the authorizers but it's better to cover that with a test for `CheckAccess()`.